### ISSUE TITLE
Renaming myexe.link

### ISSUE THREAD

**Issue by NOhs** [Trick-17/clang-build#17]
I think we should have a general dependency name for one target on another. An executable might depend on a library. But an executable might also depend on a header only target! The build-system should just know what to do:

dependence on header only library:
  - Add include files

dependence on library:
  - Add include files
  - Add linking

so how about

```toml
# Build a library
[mylib]
target_type = "sharedlibrary"
[mylib.sources]
include_directories = ["mylib/include"]
source_directories = ["mylib/src"]

[myheaderlib]
target_type = "headerlibrary"
[myheaderlib.sources]
include_directories = ["myheaderlib/include"]

# Build an executable and link the library
[myexe]
output_name = "runExe"
target_type = "executable"
dependencies = ["mylib", "myheaderlib"]
[myexe.sources]
include_directories = ["myexe/include"]
source_directories = ["myexe/src"]
[myexe.flags]
link = ["-DMYLIB_SOME_DEFINE"]
```

**Comment by GPMueller** [comment]
Agreed, the `<targetname>.link` section is not needed.

**Comment by NOhs** [comment]
Ok, I removed it in my current branch.

**Comment by NOhs** [comment]
Solved by #23
